plugins/orchestrator-form-react/src/components/OrchestratorFormWrapper.tsx

```diff
@@ -25,55 +28,126 @@ type OrchestratorFormWrapperProps = {
   numStepsInMultiStepSchema?: number;
   children: React.ReactNode;
   formData: JsonObject;
-  onSubmit: (formData: JsonObject) => void;
-  uiSchema: UiSchema<JsonObject>;
+  onChange: (formData: JsonObject) => void;
+  uiSchema: UiSchema<JsonObject, JSONSchema7>;
+};
+
+const WrapperFormPropsContext =
+  React.createContext<OrchestratorFormWrapperProps | null>(null);
+
+const useWrapperFormPropsContext = (): OrchestratorFormWrapperProps => {
+  const context = React.useContext(WrapperFormPropsContext);
+  if (context === null) {
+    throw new Error('OrchestratorFormWrapperProps not provided');
+  }
+  return context;
+};
+
+const FormComponent = (decoratorProps: FormDecoratorProps) => {
+  const props = useWrapperFormPropsContext();
+  const {
+    numStepsInMultiStepSchema,
+    uiSchema,
+    schema,
+    onChange,
+    formData,
+    children,
+  } = props;
+  const [extraErrors, setExtraErrors] = React.useState<
+    ErrorSchema<JsonObject> | undefined
+  >();
+  const isMultiStep = numStepsInMultiStepSchema !== undefined;
+  const { handleNext, activeStep, handleValidateStarted, handleValidateEnded } =
+    useStepperContext();
+  const [validationError, setValidationError] = React.useState<
+    Error | undefined
+  >();
+  const validator = useValidator(isMultiStep);
+  const getActiveKey = () => {
+    if (!isMultiStep) {
+      return undefined;
+    }
+    return Object.keys(schema.properties || {})[activeStep];
+  };
+
+  const onSubmit = async (_formData: JsonObject) => {
+    setExtraErrors(undefined);
+    let _extraErrors: ErrorSchema<JsonObject> | undefined = undefined;
+    let _validationError: Error | undefined = undefined;
+    if (decoratorProps.getExtraErrors) {
+      try {
+        handleValidateStarted();
+        _extraErrors = await decoratorProps.getExtraErrors(formData);
+        const activeKey = getActiveKey();
+        setExtraErrors(
+          activeKey && _extraErrors?.[activeKey]
+            ? (_extraErrors[activeKey] as ErrorSchema<JsonObject>)
+            : _extraErrors,
+        );
+      } catch (err) {
+        _validationError = err as Error;
+      } finally {
+        handleValidateEnded();
+      }
+    }
+    setValidationError(_validationError);
+    if (
```

Comment:
How difficult is to disable the Next if there is a validation error? So the user can not issue the Submit.

Reply:
The default way RJSF form works, is that next is alwasy enabled, and the errors show after clicking on it.
This is also the way the backstage create from template works.
Otherwise, you show errors before user wants to see them, like while typing.
There are different approaches to this, but I chose to go with the consistent behavior with backstage create template.
This is also the behavior in many websites.

Reply:
thanks, it makes sense